BuildScript/BuildTasks/CreateTestConfigurations.cs

```diff
@@ -0,0 +1,180 @@
+
+    [Required]
+    public ITaskItem[] EnabledBrowsers { get; set; }
+
+    [Required]
+    public ITaskItem[] SupportedDatabaseSystems { get; set; }
+
+    [Required]
+    public ITaskItem[] EnabledDatabaseSystems { get; set; }
+
+    [Required]
+    public ITaskItem[] SupportedPlatforms { get; set; }
+
+    [Required]
+    public ITaskItem[] EnabledPlatforms { get; set; }
+
+    [Required]
+    public ITaskItem[] SupportedExecutionRuntimes { get; set; }
+
+    [Required]
+    public ITaskItem[] EnabledExecutionRuntimes { get; set; }
+
+    [Required]
+    public ITaskItem[] SupportedTargetRuntimes { get; set; }
+
+    [Required]
+    public ITaskItem[] EnabledTargetRuntimes { get; set; }
+
+    [Required]
+    public ITaskItem[] EnabledConfigurationIDs { get; set; }
+
+    [Output]
+    public ITaskItem[] Output { get; set; }
+
+    public override bool Execute ()
+    {
+      var testConfigurationsFactory = new TestConfigurationsFactory (
+          supportedBrowsers: GetMetadataValueCollection (SupportedBrowsers, MetadataValueConstants.NoBrowser),
+          supportedDatabaseSystems: GetMetadataValueCollection (SupportedDatabaseSystems, MetadataValueConstants.NoDB),
+          supportedPlatforms: GetMetadataValueCollection (SupportedPlatforms),
+          supportedConfigurationIDs: s_supportedConfigurationIDs,
+          supportedExecutionRuntimes: GetMetadataValueDictionary (
+              SupportedExecutionRuntimes,
+              MetadataValueConstants.EnforcedLocalMachine,
+              MetadataValueConstants.LocalMachine),
+          supportedTargetRuntimes: GetMetadataValueCollection (SupportedTargetRuntimes));
+
+      var testConfigurationsFilter = new TestConfigurationsFilter (
+          supportedBrowsers: GetMetadataValueCollection (SupportedBrowsers, MetadataValueConstants.NoBrowser),
+          enabledBrowsers: GetMetadataValueCollection (EnabledBrowsers),
+          supportedDatabaseSystems: GetMetadataValueCollection (SupportedDatabaseSystems, MetadataValueConstants.NoDB),
+          enabledDatabaseSystems: GetMetadataValueCollection (EnabledDatabaseSystems),
+          supportedPlatforms: GetMetadataValueCollection (SupportedPlatforms),
+          enabledPlatforms: GetMetadataValueCollection (EnabledPlatforms),
+          supportedExecutionRuntimes: GetMetadataValueDictionary (
+              SupportedExecutionRuntimes,
+              MetadataValueConstants.EnforcedLocalMachine,
+              MetadataValueConstants.LocalMachine),
+          enabledExecutionRuntimes: GetMetadataValueCollection (EnabledExecutionRuntimes),
+          supportedTargetRuntimes: GetMetadataValueCollection (SupportedTargetRuntimes),
+          enabledTargetRuntimes: GetMetadataValueCollection (EnabledTargetRuntimes),
+          supportedConfigurationIDs: s_supportedConfigurationIDs,
+          enabledConfigurationIDs: GetMetadataValueCollection (EnabledConfigurationIDs));
+
+      var output = new List<ITaskItem>();
```

Comment:
I wonder if we should extract the loop's body and rewrite the loop as a Select inside a try-catch-block. line 105 could throw an exception. the try-catch on line 116 could be removed.

Reply:
I just played around with it but I'm struggling with the metadata extraction from the TaskItem, the first thing I could think of was projecting them into a collection of tuples and then working from there, it just looks like a cluster🚚 and I'm not sure if I should pursue this approach further

```
      var testConfigurations = Input
          .Select (
              x => (
                  x.ItemSpec,
                  TestConfiguration: x.GetMetadata (c_testConfigurationMetadataName),
                  TestSetupBuildFile: x.GetMetadata (c_testSetupBuildFileMetadataName)))
          .SelectMany (
              x => testConfigurationsFactory.CreateTestConfigurations (
                  x.ItemSpec,
                  x.TestConfiguration.Split (new[] { ";" }, StringSplitOptions.RemoveEmptyEntries).Select (y => y.Trim()).ToArray(),
                  x.TestSetupBuildFile)).ToArray();

      Output = testConfigurationsFilter.GetFilteredTestConfigurations (testConfigurations).Select (CreateTestConfigurationTaskItem).ToArray();
```